### PR TITLE
Add video availability/visibility key to info json

### PR DESCRIPTION
**Pull request by colethedj**
### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [X] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Add video availability key to the info json

As requested by https://github.com/yt-dlp/yt-dlp/issues/185

Also related: https://github.com/ytdl-org/youtube-dl/issues/25631

TODO:
- documentation
- testing
- 'availability' vs 'visibility' for the key name?

### CODE REVIEW

**Comment by pukkandan** [comment]
Closes https://github.com/ytdl-org/youtube-dl/issues/25631 https://github.com/yt-dlp/yt-dlp/issues/185

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -2036,7 +2036,14 @@ def feed_entry(name):
             or parse_duration(search_meta('duration'))
         is_live = video_details.get('isLive')
         owner_profile_url = microformat.get('ownerProfileUrl')
-
+        if not isinstance(video_details.get('isPrivate'), bool) or not isinstance(microformat.get('isUnlisted'), bool):
+            availability = None
+        elif video_details.get('isPrivate'):
+            availability = "private"
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:2042]: we use single quotes everywhere

File: yt_dlp/extractor/youtube.py [hunk 2]
```diff
@@ -2036,7 +2036,14 @@ def feed_entry(name):
             or parse_duration(search_meta('duration'))
         is_live = video_details.get('isLive')
         owner_profile_url = microformat.get('ownerProfileUrl')
-
+        if not isinstance(video_details.get('isPrivate'), bool) or not isinstance(microformat.get('isUnlisted'), bool):
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:2039]: line is too long. Split into 2 for readablity

**Review by pukkandan — COMMENTED** [review]
We'll finalize this tonight. Some comments to myself

TODO:
* ~~generalize and~~ document in common.py and Readme.md
* Add this field for playlists if possible
* Label premium ~~and member-only videos~~

**Comment by cum-squirter-nerd-hurter** [comment]
I wouldn't ever mean to rush you boys, but this absolutely needs to hurry the fuck up because YouTube is about to break a shit ton of unlisted videos on July 23 which is next Friday(!!!), and this feature would be the only practical way to sort through playlists with several thousand entries or more.

**Comment by cum-squirter-nerd-hurter** [comment]
You people deserve free blowjob coupons for life. This fucking thing sorted through several thousand YouTube bookmarks and found all the unlisted videos before Google has the chance to fuck most of them irreversibly on July 23. Everyone download yours before then with the command `--match-filter "availability = unlisted"`

**Pull request merged**